google/cloud/sqlalchemy_spanner/sqlalchemy_spanner.py

```diff
@@ -48,7 +48,8 @@
 @listens_for(Pool, "reset")
 def reset_connection(dbapi_conn, connection_record):
     """An event of returning a connection back to a pool."""
-    dbapi_conn.connection.staleness = None
+    if getattr(dbapi_conn.connection, "staleness") is not None:
```

Comment:
```suggestion
    if getattr(dbapi_conn.connection, "staleness", None) is not None:
```
Otherwise it raises:
```
In [1]: foo = 'foo'

In [2]: getattr(foo, 'encode')
Out[2]: <function str.encode(encoding='utf-8', errors='strict')>

In [3]: getattr(foo, 'asdf')
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-3-4a71e18ff97c> in <module>
----> 1 getattr(foo, 'asdf')

AttributeError: 'str' object has no attribute 'asdf'

In [4]: getattr(foo, 'asdf', None)

In [5]: 
```

Reply:
With this change, both failing examples in #186 pass 😄